stage1/init/common/pod.go

```diff
@@ -380,49 +388,39 @@ func findBinPath(p *stage1commontypes.Pod, appName types.ACName, app types.App,
 // DeviceAllow= line.
 func generateDeviceAllows(root string, appName types.ACName, mountPoints []types.MountPoint, mounts []mountWrapper, vols map[types.ACName]types.Volume, uidRange *user.UidRange) ([]string, error) {
 	var devAllow []string
-	var toShift []string
 
 	rktVolumeLinksPath := filepath.Join(root, "rkt", "volumes")
 	if err := os.MkdirAll(rktVolumeLinksPath, 0600); err != nil {
 		return nil, err
 	}
-	toShift = append(toShift, rktVolumeLinksPath)
+	if err := shiftFiles([]string{rktVolumeLinksPath}, uidRange); err != nil {
+		return nil, err
+	}
 
 	for _, m := range mounts {
 		v := vols[m.Volume]
-		if v.Kind == "host" {
-			if fileutil.IsDeviceNode(v.Source) {
-				mode := "r"
-				if !IsMountReadOnly(v, mountPoints) {
-					mode += "w"
-				}
-
-				tgt := filepath.Join(common.RelAppRootfsPath(appName), m.Path)
-				// the DeviceAllow= line needs the link path in /dev/.rkt/
-				linkRel := filepath.Join("/dev/.rkt", v.Name.String())
-				// the real link should be in /rkt/volumes for now
-				link := filepath.Join(rktVolumeLinksPath, v.Name.String())
-
-				err := os.Symlink(tgt, link)
-				switch {
-				case err == nil:
-					toShift = append(toShift, link)
-				case os.IsExist(err):
-					// it already exists, do nothing
-				default:
-					return nil, err
-				}
-
-				devAllow = append(devAllow, linkRel+" "+mode)
-			}
+		if v.Kind != "host" {
+			continue
 		}
-	}
+		if fileutil.IsDeviceNode(v.Source) {
+			mode := "r"
+			if !IsMountReadOnly(v, mountPoints) {
+				mode += "w"
+			}
 
-	if uidRange.Shift != 0 && uidRange.Count != 0 {
-		for _, f := range toShift {
-			if err := os.Chown(f, int(uidRange.Shift), int(uidRange.Shift)); err != nil {
+			tgt := filepath.Join(common.RelAppRootfsPath(appName), m.Path)
+			// the DeviceAllow= line needs the link path in /dev/.rkt/
+			linkRel := filepath.Join("/dev/.rkt", v.Name.String())
+			// the real link should be in /rkt/volumes for now
+			link := filepath.Join(rktVolumeLinksPath, v.Name.String())
+
+			err := os.Symlink(tgt, link)
+			// if the link already exists, we don't need to do anything
+			if err != nil && !os.IsExist(err) {
```

Comment:
Is `os.IsExist()` possible? `os.Symlink()` only returns `LinkError`:

https://golang.org/src/os/file_unix.go?s=10131:10174#L360